wordpress-mobile/WordPress-Android
# Update search logic to include blog name and status

Fixes #19484

This PR includes additional "blog name" and "status" properties to filter my domains list by. In addition, I moved the filter logic from the UI layer to the ViewModel to make it more efficient. The previous implementation caused unnecessary invoking of filtering while scrolling the list. Composables should just render data to avoid performance issues caused by repeated calculations while recomposition.

To test:
- Make sure you enabled `domain_management` feature flag
- If your account doesn't have a list of domains you can use a mocked list via sandboxing.  (Details here - p1692765970283879-slack-C05NS0YV7HS) Unfortunately, you can only search by domain and status using the mocked domains.
- Go to Me -> Domains
- [ ] Make sure you can search locally for domains by blog name and domain status

## Regression Notes
1. Potential unintended areas of impact
All Domains screen

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:


UI Changes testing checklist:
